src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -699,43 +706,35 @@ public void deregisterWorker(String workerName) throws IOException {
 
   @SuppressWarnings("ConstantConditions")
   @Override
-  public synchronized Set<String> getStorageWorkers() throws IOException {
-    if (storageWorkersDeadline == null || storageWorkersDeadline.isExpired()) {
-      synchronized (storageWorkerSet) {
-        Set<String> newWorkerSet = client.call(this::fetchAndExpireStorageWorkers);
-        storageWorkerSet.clear();
-        storageWorkerSet.addAll(newWorkerSet);
-      }
-      storageWorkersDeadline = Deadline.after(workerSetMaxAge, SECONDS);
-    }
-    return new HashSet<>(storageWorkerSet);
+  public Set<String> getStorageWorkers() throws IOException {
+    refreshStorageWorkersIfExpired();
+    return new HashSet<>(storageWorkers.keySet());
```

Comment:
It was missing before, but add a comment here that indicates that we're making a *copy* of the set for return, and not providing access to the shared set.

Reply:
Added comment on method.